### PR TITLE
Add visualization map of embeddings

### PR DESCRIPTION
**Pull request by jilljenn**
It is currently located at https://mangaki.fr/map/

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
I do not see the source code of the map in this PR, am I wrong?

File: mangaki/mangaki/utils/viz.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# SPDX-FileCopyrightText: 2014, Mangaki Authors
```
[the remark this answers is not in the document]
Comment (jilljenn) [reply mangaki/mangaki/utils/viz.py:1]: I don't even remember where this comes from, but I think it was NLnet

**Comment by jilljenn** [comment]
> I do not see the source code of the map in this PR, am I wrong?

You can see it live https://mangaki.fr/map/ but I may create a separate repo for it

**Comment by jilljenn** [comment]
- Adding a `api/user/position/<algo_name>` route that returns the position of the user and their friends (if private, friend relationship has to be reciprocal to be displayed)
- `fit_algo` management command has a `--viz_only` parameter that only stores embeddings instead of retraining
- Embeddings are stored in `VIZ_ROOT` (because of that I had to fix the path for several tests)
- Fitting an algo also stores embeddings by default, that are used to compute the user positions

File: mangaki/mangaki/api/user.py [hunk 2]
```diff
@@ -90,5 +97,42 @@ def export_user_data(request: Request):
...
+    """
+    rated_works = current_user_ratings(request)
+    df_mine = pd.DataFrame(rated_works.items(), columns=('work_id', 'choice'))
+    df_mine['user_id'] = request.user.id if not request.user.is_anonymous else -1
+    df_item_pos = pd.DataFrame(
+        get_2d_embeddings(algo_name)['works']).set_index('work_id')
+    available_works = df_item_pos.index
+
+    if not request.user.is_anonymous:
+        friend_ratings = friend_ratings(request)
```
Comment (Akulen) [inline mangaki/mangaki/api/user.py:117]: Isn't this missing friend_ids argument?

Comment (jilljenn) [reply mangaki/mangaki/api/user.py]: Indeed and also it was overriding the function with the variable :roll_eyes:

**Review by Akulen — APPROVED** [review]

**Pull request merged**